client/driver/executor/test_harness_test.go

```diff
@@ -14,6 +15,57 @@ import (
 	"github.com/hashicorp/nomad/nomad/structs"
 )
 
+func TestMain(m *testing.M) {
+	switch os.Getenv("TEST_MAIN") {
```

Comment:
Please add a comment that we switch the behavior depending on the environment variable and why we do this.